modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationSpi.java

```diff
@@ -3262,17 +3277,24 @@ protected GridCommunicationClient createTcpClient(ClusterNode node, int connIdx)
         GridCommunicationClient client = null;
         IgniteCheckedException errs = null;
 
-        int connectAttempts = 1;
-
-        for (InetSocketAddress addr : addrs) {
-            long connTimeout0 = connTimeout;
-
-            int attempt = 1;
+        long totalTimeout;
 
-            IgniteSpiOperationTimeoutHelper timeoutHelper = new IgniteSpiOperationTimeoutHelper(this,
-                !node.isClient());
+        if (failureDetectionTimeoutEnabled())
+            totalTimeout = node.isClient() ? clientFailureDetectionTimeout() : failureDetectionTimeout();
+        else {
+            totalTimeout = ExponentialBackoffTimeoutStrategy.totalBackoffTimeout(
```

Comment:
maxWaitTime should be used. Static method brokes strategy pattern. totalTimeout  should be calculated inside strategy ctor.